cdap-api/src/main/java/co/cask/cdap/api/ProgramState.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.api;
+
+import javax.annotation.Nullable;
+
+/**
+ * Class to represent the state of the program.
+ */
+public class ProgramState {
+  private final ProgramStatus status;
+  private final String failureCause;
```

Comment:
Then, call it `failureInfo` because `failureCause` is commonly having type `Throwable`.